OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromCDLogic.cs

```diff
@@ -64,6 +80,50 @@ void CheckForDisk()
 			}
 		}
 
+		void InstallTFD(string source)
+		{
+			backButton.IsDisabled = () => true;
+			retryButton.IsDisabled = () => true;
+			insertDiskContainer.IsVisible = () => false;
+			installingContainer.IsVisible = () => true;
+			progressBar.Percentage = 0;
+
+			new Thread(() =>
+			{
+				using (var cab_ex = new InstallShieldCABExtractor(source))
```

Comment:
Style nit: we don't use underscores in variable names. `cab_ex` &rarr; `cabExtractor`?